apps/ecosystem/src/ecosystem.repository.ts

```diff
@@ -328,12 +329,157 @@ export class EcosystemRepository {
+        where: {
+          orgId
+        }
+      });
+      return agentDetails;
+
+    } catch (error) {
+      this.logger.error(`Error in getting getAgentEndPoint for the ecosystem: ${error.message} `);
+      throw error;
+    }
+  }
+
+  /**
+     * Description: Get getAgentEndPoint by orgId
+     * @param orgId 
+     * @returns Get getAgentEndPoint details
+     */
+  // eslint-disable-next-line camelcase
+  async getEcosystemLeadDetails(): Promise<ecosystem_orgs> {
+    try {
+      const ecosystemRoleDetails = await this.prisma.ecosystem_roles.findFirst({
+        where: {
+          name: EcosystemRoles.ECOSYSTEM_LEAD
+        }
+      });
+      const ecosystemLeadDetails = await this.prisma.ecosystem_orgs.findFirst({
+        where: {
+          ecosystemRoleId: ecosystemRoleDetails.id
+        }
+      });
+      return ecosystemLeadDetails;
+
+    } catch (error) {
+      this.logger.error(`Error in getting ecosystem lead details for the ecosystem: ${error.message} `);
+      throw error;
+    }
+  }
+
+  /**
+   * Get platform config details
+   * @returns 
+   */
+  // eslint-disable-next-line camelcase
+  async getPlatformConfigDetails(): Promise<platform_config> {
+    try {
+
+      return this.prisma.platform_config.findFirst();
+
+    } catch (error) {
+      this.logger.error(`Error in getting getPlatformConfigDetails for the ecosystem - error: ${JSON.stringify(error)}`);
+      throw new InternalServerErrorException(error);
+    }
+  }
+
+  async storeTransactionRequest(
+    schemaTransactionResponse: SchemaTransactionResponse
+  ): Promise<object> {
+    try {
+      const { endorserDid, authorDid, requestPayload, status, ecosystemOrgId } = schemaTransactionResponse;
+      return await this.prisma.endorsement_transaction.create({
+        data: {
+          endorserDid,
+          authorDid,
+          requestPayload,
+          status,
+          ecosystemOrgId,
+          responsePayload: ''
+        }
+      });
+    } catch (error) {
+      this.logger.error(`error: ${JSON.stringify(error)}`);
+      throw new InternalServerErrorException(error);
+    }
+  }
+  // eslint-disable-next-line camelcase
+  async getEcosystemOrgDetailsbyId(orgId: string): Promise<ecosystem_orgs> {
+    try {
+      const ecosystemLeadDetails = await this.prisma.ecosystem_orgs.findFirst({
+        where: {
+          orgId
+        }
+      });
+      return ecosystemLeadDetails;
+
+    } catch (error) {
+      this.logger.error(`Error in getting ecosystem lead details for the ecosystem: ${error.message} `);
+      throw error;
+    }
+  }
+  // eslint-disable-next-line camelcase
+  async getEndorsementTransactionById(endorsementId: string): Promise<endorsement_transaction> {
+    try {
+      const ecosystemLeadDetails = await this.prisma.endorsement_transaction.findFirst({
+        where: {
+          id: endorsementId,
+          status: endorsementTransactionStatus.REQUESTED
```

Comment:
Can you pass the status `REQUESTED` dynamic like we can pass `where` condition data as payload to the function

Reply:
I changed that, it will be reflected in the upcoming PR